src/lib/types/Web3FunctionContext.ts

```diff
@@ -37,3 +57,22 @@ export interface Web3FunctionContext {
 export interface Web3FunctionEventContext extends Web3FunctionContext {
   log: Log;
 }
+export interface Web3FunctionSuccessContext extends Web3FunctionContext {
+  transactionHash?: string;
+}
+export interface Web3FunctionEventSuccessContext
+  extends Web3FunctionEventContext {
+  transactionHash?: string;
+}
+
+export type FailReason =
+  | "InsufficientFunds"
+  | "SimulationFailed"
+  | "ExecutionReverted";
+
+export interface Web3FunctionFailContext extends Web3FunctionContext {
+  reason: FailReason;
```

Comment:
Should we pass back the transaction callData that failed?
So that user can eventually decode it and know which operation failed if it needs to retry or flag it somehow?